clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StatusReportAdapter.java

```diff
@@ -231,32 +208,28 @@ public void recordSubPartitionStatus(
                 statusRecordMap.get(versionAwareStatus));
           }
           partitionConsumptionState.recordSubPartitionStatus(versionAwareStatus);
-          statusRecordMap.get(versionAwareStatus).add(subPartitionIndex);
-          updatedCount = counter.incrementAndGet();
-        } else {
-          updatedCount = counter.get();
         }
+        updatedCount = updateResult ? counter.incrementAndGet() : counter.get();
+      }
+      if (updatedCount == amplificationFactor) {
+        maybeReportStatus(status, versionAwareStatus, report, logStatus);
       }
-      maybeReportStatus(status, versionAwareStatus, updatedCount, report, logStatus);
     }
 
     private void maybeReportStatus(
         SubPartitionStatus status,
         String versionAwareStatus,
-        int updatedCount,
         Runnable report,
         boolean logStatus) {
-      if (updatedCount == amplificationFactorAdapter.getAmplificationFactor()) {
-        // This is a safeguard to make sure we only report exactly once for each status.
-        if (statusReportMap.get(versionAwareStatus).compareAndSet(false, true)) {
-          if (logStatus) {
-            LOGGER.info("Reporting status {} for user partition: {}.", versionAwareStatus, userPartition);
-          }
-          report.run();
-          if (status.equals(COMPLETED)) {
-            amplificationFactorAdapter
-                .executePartitionConsumptionState(userPartition, PartitionConsumptionState::completionReported);
-          }
+      // This is a safeguard to make sure we only report exactly once for each status.
+      if (statusReportMap.get(versionAwareStatus).compareAndSet(false, true)) {
```

Comment:
This tripped me up, because `statusReportMap` looks so similar to `statusRecordMap`... any chance we could find more intuitive names for these maps?